apps/system/lockscreen/js/widgets/clock/lockscreen_clock_widget.js

```diff
@@ -27,13 +27,16 @@
     var f = new navigator.mozL10n.DateTimeFormat();
     var _ = navigator.mozL10n.get;
 
+    /* We use the span tag only as a marker to remove the am/pm component of
+     * the string as we don't show it in the lockscreen. */
     var timeFormat = this._timeFormat.replace('%p', '<span>%p</span>');
     var dateFormat = _('longDateFormat');
 
-    var timeHTML = f.localeFormat(now, timeFormat);
+    var timeText =
+      f.localeFormat(now, timeFormat).replace(/\s<span>.*<\/span>/, '');
```

Comment:
This code becomes kinda puzzled for me: you add the span 2 lines ago, and now you remove it here. May you create the `timeText` more straightly? And, did you check the UI about 12/24 hours on LockScreen is unaffected by this patch? I may ask a UI review after code review.